File: sketch0218/loong/UserCtrl/Revertable/Revertable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UserCtrl
{
    public class Revertable
    {
        public Revertable()
        {

        }

        public virtual bool isValid()
        {
            return true;
        }

        public virtual void undo()
        {

        }

        public virtual void redo()
        {

        }
    }
}

File: sketch0218/loong/sketch/src/Command/Revertable/Revertable.cpp
#include "sketch/Command/Revertable/Revertable.h"

namespace sketch
{
Revertable::Revertable(Define::RevertableType type) : m_type(type) {}

bool Revertable::isValid() { return true; }

bool Revertable::undo() { return true; }

bool Revertable::redo() { return true; }
}// namespace sketch
